src/test/unit_tests/test_nsfs_access.js

```diff
@@ -99,6 +110,28 @@ mocha.describe('new tests check', async function() {
             assert.equal(err.code, 'EACCES');
         }
     });
+
+    mocha.it('NON ROOT 3 with suplemental group - success', async function() {
+        await nb_native().fs.mkdir(NON_ROOT1_FS_CONFIG, full_path_non_root1, 0o770);
+        //TODO on mac new directories are created with the parents directory GID and not with the process GID. manually change the gid
+        fs.promises.chown(full_path_non_root1, NON_ROOT1_FS_CONFIG.uid, NON_ROOT1_FS_CONFIG.gid);
+        //non root3 has non-root1 group as supplemental group, so it should succeed
+        const non_root_entries = await nb_native().fs.readdir(NON_ROOT3_FS_CONFIG, full_path_non_root1);
+        assert.equal(non_root_entries && non_root_entries.length, 0);
+    });
+
+    mocha.it('NON ROOT 3 suplemental group without the files gid - failure', async function() {
+        //TODO on mac new directories are created with the parents directory GID and not with the process GID. manually change the gid
+        fs.promises.chown(full_path_non_root1, NON_ROOT2_FS_CONFIG.uid, NON_ROOT2_FS_CONFIG.gid);
```

Comment:
Why is it here and not after the `nb_native().fs.mkdir`?
Why the `fs.promises.chown` is on `full_path_non_root1` and not `full_path_non_root2`?

Reply:
changed it, thanks